falcon server - ioc (#59)
deity-io/falcon-platform-docs: docs/platform/server/api.md

---
title: Falcon Server API
---

> This section provides an API description for both packages - `@deity/falcon-server`
> and `@deity/falcon-server-env`

## Dependency Inversion

In order to bring more flexibility in dependencies instance lifetime management, Falcon Server internally implements Dependency Inversion pattern via [Iversify JS](http://inversify.io/) framework.

> It means that Falcon Server internals, including extension components (Api DataSources, Extensions end Endpoints) are registered in Inversify container. Even they are setted up via configuration files.

Reviewer: we could add more info about some internal things that we did, like getting ApiDataSources by using registered keywords, like `Config`, `EventEmitter2`, containers (Api, Endpoint, Extension), entries by type - `ApiDataSource`, `{name}DataSource`, components and so on. These things are highly useful and important to know.

> In order to override existing or register new dependency in Falcon Server you need to override `configure` method of Falcon Server:

Reviewer 2: ```suggestion
In order to override or register a dependency in Falcon Server you need to override the `configure` method in Falcon Server:
```
